src/core/src/preprocess/pre_post_process.cpp

```diff
@@ -486,9 +486,11 @@ std::shared_ptr<Function> PrePostProcessor::build() {
         }
 
         auto node = nodes[0];
-
+        if (node.get_partial_shape() != param->get_partial_shape()) {
+            tensor_data_updated = true;  // Trigger revalidation if input parameter shape is changed
+        }
         // Check final shape
-        OPENVINO_ASSERT(node.get_partial_shape().refines(param->get_partial_shape()),
+        OPENVINO_ASSERT(node.get_partial_shape().compatible(param->get_partial_shape()),
```

Comment:
This line fixes original issue, but output becomes {?,?,?,?} instead of {?,3?,?} thus I decided to change reverse_channels in this PR as well